torch/distributed/tensor/_api.py

```diff
@@ -211,6 +212,192 @@ def backward(ctx, grad_output: "DTensor"):  # type: ignore[override]
         return grad_output.to_local(), None, None, None, None, None
 
 
+def _prepare_placements_and_shard_order(
+    device_mesh: DeviceMesh,
+    tensor_rank: int,
+    placements: Optional[Sequence[Placement]] = None,
+    shard_order: Optional[dict[int, list[int | str]]] = None,
+) -> tuple[
+    tuple[Placement, ...],  # update placements in tuple
+    tuple[tuple[int, ...], ...],  # updated shard_order in tuple
+]:
+    """
+    Helper function to prepare placements and shard_order for
+    ``distribute_tensor`` and DTensor ``redistribute``.
+
+    Args:
+        device_mesh (:class:`DeviceMesh`): DeviceMesh to place the tensor.
+        tensor_rank (int): The rank (number of dimensions) of the tensor to be
+            distributed or redistributed.
+        placements (List[:class:`Placement`], optional): The placements that
+            describe how to place the local torch.Tensor on DeviceMesh. Must
+            have the same number of elements as ``device_mesh.ndim``.
+        shard_order (Optional[Dict[int, List[int | str]]], optional): Specifies
+            the mapping of tensor dimensions to the order of device mesh
+            dimensions they are sharded over. Each key is a tensor dimension
+            (can be negative for reverse indexing), and its value is a list
+            specifying the device mesh dimensions (as integers or strings) that
+            the tensor dimension is sharded across, in order. If not specified,
+            a default sharding order is used.
```

Comment:
You gotta say what the default is. But investing heavily in this docblock is also a bit weird, because this is private API; really you should focus on explaining well in the real documentation.